crates/table/src/bflatn_to_bsatn_fast_path.rs

```diff
@@ -0,0 +1,424 @@
+//! This module implements a fast path for serializing certain types from BFLATN to BSATN.
+//!
+//! The key insight is that a majority of row types in a game will have a known fixed length,
+//! with no variable-length members.
+//! BFLATN is designed with this in mind, storing fixed-length portions of rows inline,
+//! at the expense of an indirection to reach var-length columns like strings.
+//! A majority of these types will also have a fixed BSATN length,
+//! but note that BSATN stores sum values (enums) without padding,
+//! so row types which contain sums may not have a fixed BSATN length
+//! if the sum's variants have different "live" unpadded lengths.
+//!
+//! For row types with fixed BSATN lengths, we can reduce the BFLATN -> BSATN conversion
+//! to a series of `memcpy`s, skipping over padding sequences.
```

Comment:
I added:
```rust
//! This is potentially much faster than the more general  [`crate::bflatn_from::serialize_row_from_page`],
//! which traverses a [`RowTypeLayout`] and dispatches on the type of each column.
```
If you'd like more, let me know.

Reply:
Maybe a little more.

The reason this is faster, to my understanding, which may be incomplete, is that we can memcpy each bfatn field into a statically sized bsatn buffer. We can't memcpy in the general case because, in general, the size of the bsatn buffer cannot be computed statically.

Is this accurate?

> skipping over padding sequences

This is also a little confusing because I can't tell if you're implying that this contributes to the speed up because we can't skip over padding in the general case, or if this is just a constraint on the memcpy. I believe it's the latter, but wasn't entirely sure.

Reply:
What I'm trying to say here is that we can copy multiple fields in a single `memcpy` if they're stored without any intermediate padding in the BFLATN. So the fast path can do a single memcpy for `(u64, u64, u64)`, whereas the slow path will do 3 dispatches, 3 reads and 3 `serialize_u64`s.

Reply:
Added:
```rust
//! For example, to serialize a row of type `(u64, u64, u32, u64)`,
//! [`bflatn_from`] will do four dispatches, three calls to `serialize_u64` and one to `serialize_u32`.
//! This module will make 2 `memcpy`s (or actually, `<[u8]>::copy_from_slice`s):
//! one of 20 bytes to copy the leading `(u64, u64, u32)`, which contains no padding,
//! and then one of 8 bytes to copy the trailing `u64`, skipping over 4 bytes of padding in between.
```